### ISSUE TITLE
says "nothing to clear" and requires a php-restart to clear

### ISSUE THREAD

**Issue by vesper8** [appstract/laravel-opcache#65]
I've been using this for months and it was working fine.. but today I had to install a new composer package. I deployed my new composer.json/.lock and ran composer install --no-dev on production

then I started having this Uncaught ReflectionException: Class view does not exist error.. it took me a while to figure out the pa opcache:clear command was failing

I tried running it and it always said "Opcode cache: Nothing to clear" .. what ?? =/

I checked the options for that command and don't see a --force.. maybe there should be a --force command? I'd feel safer with it.. not sure why this happened but it caused some downtime and made me panic a bit

Finally I did a sudo service php7.2-fpm restart and that cleared the error.. clearly it wasn't seeing the new composer changes

**Comment by ovanschie** [comment]
There is no way to force it. This message is shown when it could not clear something. Probably something wrong with your PHP or Opcache config

**Comment by vesper8** [comment]
@ovanschie I can reproduce this every time now, and yet it was working before.

I never changed any of the settings. The APP_URL is set correctly.

This is the output of opcache:config

```
+---------------------------------------+------------+
| option                                | value      |
+---------------------------------------+------------+
| opcache.enable                        | 1          |
| opcache.enable_cli                    |            |
| opcache.use_cwd                       | 1          |
| opcache.validate_timestamps           |            |
| opcache.validate_permission           |            |
| opcache.validate_root                 |            |
| opcache.inherited_hack                | 1          |
| opcache.dups_fix                      |            |
| opcache.revalidate_path               |            |
| opcache.log_verbosity_level           | 1          |
| opcache.memory_consumption            | 512.00 MB  |
| opcache.interned_strings_buffer       | 64         |
| opcache.max_accelerated_files         | 30000      |
| opcache.max_wasted_percentage         | 0.05       |
| opcache.consistency_checks            | 0          |
| opcache.force_restart_timeout         | 180        |
| opcache.revalidate_freq               | 2          |
| opcache.preferred_memory_model        |            |
| opcache.blacklist_filename            |            |
| opcache.max_file_size                 | 0          |
| opcache.error_log                     |            |
| opcache.protect_memory                |            |
| opcache.save_comments                 | 1          |
| opcache.enable_file_override          |            |
| opcache.optimization_level            | 2147467263 |
| opcache.lockfile_path                 | /tmp       |
| opcache.file_cache                    |            |
| opcache.file_cache_only               |            |
| opcache.file_cache_consistency_checks | 1          |
+---------------------------------------+------------+
```

I run opcache:clear at the end of my Envoy deployment script, and everytime it says "nothing to clear" And the site is broken until I restart php

You are pretty quick to close this issue I think.. the issue might have something to do with php 7.2 or a conflict with an updated library?

Some help debugging would be appreciated

**Comment by ovanschie** [comment]
Tried on a clean 7.2 install, works as expected. Will reopen so maybe someone can help.

Will debug a bit more myself soon.

**Comment by vesper8** [comment]
I'm still having this problem.. I've re-ordered my deployment script and now it says "opcache cache cleared" and yet I STILL have to run  'restart sudo service php7.2-fpm restart' before the new changes are available

maybe this is because of the order in which I clear my local laravel cache.. here's that part of deployment script.. do you see anything wrong with it?

```
# forceClearCache
rm -f {{ $release }}/bootstrap/cache/compiled.php {{ $release }}/bootstrap/cache/config.php {{ $release }}/bootstrap/cache/routes.php {{ $release }}/bootstrap/cache/services.json {{ $release }}/bootstrap/cache/packages.php

php {{ $release }}/artisan config:cache
php {{ $release }}/artisan route:clear
php {{ $release }}/artisan route:scan
php {{ $release }}/artisan view:clear
php {{ $release }}/artisan clear-compiled

cd {{ $release }}
composer dump-auto

php {{ $release }}/artisan opcache:clear
php {{ $release }}/artisan cloudflare:cache:purge

php {{ $release }}/artisan cache:clear
php {{ $release }}/artisan responsecache:flush

php {{ $release }}/artisan debugbar:clear

php {{ $release }}/artisan horizon:terminate

echo 'All Cleared'
```

**Comment by ovanschie** [comment]
Show me the rest of your deploy script

**Comment by vesper8** [comment]
sure.. I hope this isn't too hard to follow

```
@setup
require __DIR__.'/vendor/autoload.php';
$dotenv = new Dotenv\Dotenv(__DIR__);
try {
$dotenv->load();
$dotenv->required(['DEPLOY_SERVER', 'DEPLOY_REPOSITORY', 'DEPLOY_PATH'])->notEmpty();
} catch ( Exception $e )  {
echo $e->getMessage();
}

$server = getenv('DEPLOY_SERVER');
$repo = getenv('DEPLOY_REPOSITORY');
$path = getenv('DEPLOY_PATH');
$branch = getenv('DEPLOY_BRANCH');
$slack = getenv('DEPLOY_SLACK_WEBHOOK');

if ( substr($path, 0, 1) !== '/' ) throw new Exception('Careful - your deployment path does not begin with /');

$date = ( new DateTime )->format('YmdHis');
$env = isset($env) ? $env : "production";
$branch = isset($branch) ? $branch : "master";
$path = rtrim($path, '/');
$release = $path.'/'.$date;
@endsetup

@servers(['web' => $server])


@story('deploy')
deployment_start
deployment_links
deployment_composer
deployment_cache
deployment_finish
deployment_option_cleanup
@endstory

@task('deployment_start')
cd {{ $path }}
echo "Deployment ({{ $date }}) started"
git clone {{ $repo }} --branch={{ $branch }} --depth=1 -q {{ $release }}
echo "Repository cloned"
@endtask

@task('deployment_links')
cd {{ $path }}
rm -rf {{ $release }}/storage
ln -nfs {{ $path }}/storage {{ $release }}/storage
ln -nfs {{ $path }}/storage/app/public/media {{ $release }}/public/media
echo "Storage directories set up"
ln -nfs {{ $path }}/.env {{ $release }}/.env
echo "Environment file set up"
@endtask

@task('deployment_composer')
cd {{ $release }}
composer install --no-interaction --quiet --no-dev
@endtask

@task('deployment_cache')

# forceClearCache
rm -f {{ $release }}/bootstrap/cache/compiled.php {{ $release }}/bootstrap/cache/config.php {{ $release }}/bootstrap/cache/routes.php {{ $release }}/bootstrap/cache/services.json {{ $release }}/bootstrap/cache/packages.php

php {{ $release }}/artisan config:cache
php {{ $release }}/artisan route:clear
php {{ $release }}/artisan route:scan
php {{ $release }}/artisan view:clear
php {{ $release }}/artisan clear-compiled

cd {{ $release }}
composer dump-auto

php {{ $release }}/artisan opcache:clear
php {{ $release }}/artisan cloudflare:cache:purge

php {{ $release }}/artisan cache:clear
php {{ $release }}/artisan responsecache:flush

{{--  ./clearLogs.sh  --}}
php {{ $release }}/artisan debugbar:clear

php {{ $release }}/artisan horizon:terminate

echo 'All Cleared'
@endtask

@task('deployment_finish')
ln -nfs {{ $release }} {{ $path }}/current
echo "Deployment ({{ $date }}) finished"
@endtask

@task('deployment_option_cleanup')
cd {{ $path }}
@if ( isset($cleanup) && $cleanup )
    find . -maxdepth 1 -name "20*" -mmin +2880 | head -n 5 | xargs rm -Rf
    echo "Cleaned up old deployments"
@endif
@endtask
```

**Comment by ovanschie** [comment]
Might have something to do with the current release path??
Maybe try clearing after the symlink is created. I think you are basically clearing the old release

**Comment by vesper8** [comment]
Thanks @ovanschie !!! Looks like that was it!!
